src/nb_enum.cpp

```diff
@@ -150,6 +154,28 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
     if (!t)
         return false;
 
+    if ((t->flags & (uint32_t) type_flags::is_flag_enum) !=0) {
+        auto base = PyObject_GetAttrString((PyObject *)o->ob_type, "__base__");
+        auto basename = PyObject_GetAttrString(base, "__name__");
+        Py_ssize_t size;
+        const char* data = PyUnicode_AsUTF8AndSize(basename, &size);
+        std::string s(data, size);
```

Comment:
This is not a sufficient reason to use `std::string`; nanobind generally tries to keep a low footprint of standard-library includes, to avoid code bloat. Check the size, then use `memcmp()`.

Also, I don't think this logic is correct. You should instead check whether `Py_TYPE(o)` -- the Python type of the incoming might-be-an-enumeration-value -- matches `t->type_py` -- the Python type of the nanobind enumeration you're trying to convert to. What you have here is both much slower than that, and is foolable:
```
class Flag:
    pass

class Dummy(Flag):
    value = 42

some_nanobind_function(Dummy())  # result is treated as flags of 42!

Reply:
+1